scripts/generate_deb.sh

```diff
@@ -13,16 +13,13 @@ fi
 
 NAME="${PROJECT}_${VERSION_NO_SUFFIX}-${RELEASE}_amd64"
 
-DEPS_STR=""
-for dep in "${DEPS[@]}"; do
-   DEPS_STR="${DEPS_STR} Depends: ${dep}"
-done
 mkdir -p ${PROJECT}/DEBIAN
+chmod 0755 ${PROJECT}/DEBIAN
 echo "Package: ${PROJECT} 
 Version: ${VERSION_NO_SUFFIX}-${RELEASE}
 Section: devel
 Priority: optional
-Depends: libbz2-dev (>= 1.0), libssl-dev (>= 1.0), libgmp3-dev, build-essential, libicu-dev, zlib1g-dev
+Depends: libc6, libgcc1, libssl1.0.0, libstdc++6, libtinfo5, zlib1g
```

Comment:
> libssl1.0.0

This would seem to imply that the .deb *must* be built on 16.04. Is that intended?

Reply:
As far as I can see it's absolutely required because of libc6 and libstdc++.  In my testing, if I built on a later Ubuntu then installed on 16, the resulting binaries fail to run, complaining that version CXXABI_1.3.11 is not found.  On Ubuntu 16, the out-of-the-box libstdc++ stops with CXXABI_1.3.9.

While it is apparently possible to get CXXABI_1.3.11 on Ubuntu 16, it requires installing gcc, and all that brings with it.  It was my intent to avoid such dramatic alteration of the system if it could be avoided.

Reply:
We currently provide two .deb files though, one for 16.04 and one for 18.04. These are clearly built on the respective ubuntu versions.

The problem with putting `libssl1.0.0` here is that a deb built on 18.04 won't have the right dependencies because 1.1 is what is used on 18.04. (1.1 & 1.0 are not ABI compatible)

Reply:
That's a problem then, because there is no `libssl` meta package.  Its name on Ubuntu 16 is `libssl1.0.0` and there is no way to get only the library package by another name.  Debian dependency parsing does not support wildcards, either.  Since we're eschewing debhelper, our alternative is to hack in Ubuntu version recognition and substitute that package name depending on the result.